### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by xyproto**
* Separated types and functions into `services.go` and `supervisor.go`.
* Added three utility functions, for increased clarity of the code.
* Moved all types, constants and variables to the top, in `var ( ... )`, `const( ... )` and `type( ... )` blocks.
* Minor changes to some single-line comments (capital letter).
* All tests pass.

Reject or pull at will.

Thanks for creating suture, it looks really useful! I have always wished Go had similar mechanisms to what Erlang has to offer.

### CODE REVIEW

File: suture.go [hunk 1]
```diff
@@ -273,14 +248,12 @@ func New(name string, spec Spec) (s *Supervisor) {
 	return
 }
 
-func serviceName(service Service) (serviceName string) {
-	stringer, canStringer := service.(fmt.Stringer)
-	if canStringer {
-		serviceName = stringer.String()
-	} else {
-		serviceName = fmt.Sprintf("%#v", service)
+func serviceName(service Service) string {
```
Comment (xyproto) [inline suture.go]: I believe not using named returns and using `if (...) { return ... }; return ...` is slightly more idiomatic.

**Comment by thejerf** [comment]
I have to admit to some ambivalence... I may pick and choose some stuff. I like the float default stuff, I'm less sure about the type blocks.

In general I'm still struggling with what I consider best style there, in terms of putting types and their methods together. A similar example: About a third of my packages end up with a "types.go"; is that a good idea or an antipattern? Or, if it's inbetween, when is it which? I'm not sure.

**Comment by xyproto** [comment]
I am also conflicted as to where types is best placed: at the top or above relevant functions.

The pull request is borderline invasive, so it's completely understandable if you reject or cherry-pick from it.

Cheers!

**Comment by xyproto** [comment]
Is there a relation between 2189491098c7374353aec94db241846f49dd8858 and this pull request?

**Comment by thejerf** [comment]
On 04/23/2016 10:48 AM, Alexander F Rødseth wrote:
>
> Is there a relation between 2189491 
> <https://github.com/thejerf/suture/commit/2189491098c7374353aec94db241846f49dd8858> 
> and this pull request?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub 
> <https://github.com/thejerf/suture/pull/13#issuecomment-213752618>
>
No. I've got a bit of time to do some maintenance today but I have not 
yet gotten to the refactoring PR.

I think I will pick and choose out of it, BTW, but at least separating 
the supervisor and the service is definitely a good idea. I would 
certainly do it if I were writing this fresh today.

**Comment by thejerf** [comment]
I've separated the service, supervisor, and messages out more nicely.

I didn't actually intend to close this PR when I did, because I meant to get around to it later, which is now. Whoops.

**Pull request closed without merging**